### DISCUSSION TITLE
Format of edge_index

### DISCUSSION THREAD

**Discussion by hkim716** [pyg-team/pytorch_geometric#2880] [Q&A] [answered]
Hi Matt, I would like to draw my mesh dataset using `plotly`. 

https://plotly.com/python/3d-mesh/
Here in the **3D Mesh in Dash** example to draw a sandal, it requires `i=df.i, j=df.j, k=df.k`, format.

I think in that example, `i, j, k` means vertices of a triangular mesh, and `x, y, z` is the coordinates of each node. 
Assuming that I have triangular mesh like `edge_index`, 
I want to know how I could transform `edge_index` to the format compatible with this example.

Also, I guess `edge_index` that is used in `PyG` is called COO format with shape `[2, num_edges]`. I think it's pretty clear to understand how it works. But what is `CSR` format then?

**Comment by rusty1s** [comment, marked as answer]
There is no easy way to convert the `edge_index` representation to a mesh representation of faces. Therefore, it is best to simply keep your mesh data for visualization while converting it to the `edge_index` representation for neural network processing.

You are right that PyG represents graphs in a COO format. CSR denotes a compressed representation of this, where values are sorted by source node index, and only node-wise offsets are stored. We support this format via the [`SparseTensor` class](https://pytorch-geometric.readthedocs.io/en/latest/notes/sparse_tensor.html).
